op-e2e/actions/interop/dsl/assertions.go

```diff
@@ -1,13 +1,378 @@
 package dsl
 
 import (
+	"math/big"
+
+	"github.com/ethereum-optimism/optimism/op-e2e/actions/helpers"
+	"github.com/ethereum-optimism/optimism/op-service/eth"
+	"github.com/ethereum/go-ethereum/rpc"
+	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 )
 
-func RequireL2UnsafeNumberEquals(t require.TestingT, c *Chain, unsafeNumber uint64) {
-	require.Equal(t, unsafeNumber, c.Sequencer.L2Unsafe().Number)
+// RequireSupervisorChainHeads queries the supervisor actor for its sync status and asserts that
+// the chain's LocalUnsafe, CrossUnsafe, LocalSafe, CrossSafe, and Finalized block IDs match the
+// provided expected values. It fails the test if any assertion does not hold.
+func RequireSupervisorChainHeads(t helpers.Testing, super *SupervisorActor, chain *Chain, unsafe, crossUnsafe, localSafe, safe, finalized eth.BlockID) {
+	status, err := super.SyncStatus(t.Ctx())
+	require.NoError(t, err)
+
+	cStatus, ok := status.Chains[chain.ChainID]
+	require.True(t, ok, "chain id not present in supervisor sync status %s", chain.ChainID)
+	passed := true
+	passed = passed && assert.Equal(t, unsafe, cStatus.LocalUnsafe.ID(), "unsafe for chain id: %s", chain.ChainID)
+	passed = passed && assert.Equal(t, crossUnsafe, cStatus.CrossUnsafe, "cross unsafe for chain id: %s", chain.ChainID)
+	passed = passed && assert.Equal(t, localSafe, cStatus.LocalSafe, "local safe for chain id: %s", chain.ChainID)
+	passed = passed && assert.Equal(t, safe, cStatus.CrossSafe, "safe for chain id: %s", chain.ChainID)
+	passed = passed && assert.Equal(t, finalized, cStatus.Finalized, "finalized for chain id: %s", chain.ChainID)
+	require.True(t, passed, "One or more supervisor chain heads assertions failed")
+}
+
+// RequireUnsafeTimeOffset asserts that the difference between the sequencer's L2 unsafe block time
+// and the chain's genesis L2 time matches the provided expected offset.
+func RequireUnsafeTimeOffset(t helpers.Testing, c *Chain, timeOffset uint64) {
+	require.Equal(t, timeOffset, c.Sequencer.L2Unsafe().Time-c.RollupCfg.Genesis.L2Time)
+}
+
+// RequireL1Heads fetches the latest and finalized L1 block headers via the L1 miner client and
+// asserts that their block numbers match the expected values, treating missing finalized blocks as zero.
+func RequireL1Heads(t helpers.Testing, system *InteropDSL, latest, finalized uint64) {
+	safeHeader, err := system.Actors.L1Miner.EthClient().BlockByNumber(t.Ctx(), big.NewInt(int64(rpc.LatestBlockNumber)))
```

Comment:
```suggestion
	latestHeader, err := system.Actors.L1Miner.EthClient().BlockByNumber(t.Ctx(), big.NewInt(int64(rpc.LatestBlockNumber)))
```
because we are fetching latest block

Reply:
Fixed in 0be2205c967e3b96a2f84c7df3aca211e67a85fb.